### PR TITLE
[Fixes #2594] windows: Implement the etc_services table

### LINKED ISSUE #2594: windows: Implement the etc_services table
We have an `etc_hosts` and `etc_protocols` table on Windows (see https://github.com/facebook/osquery/issues/2566) but we are missing `etc_services`.

Follow the migration details / outline from #2566 and make `etc_services` cross-platform!

### PR DESCRIPTION
**Pull request by slester**
Fixes #2594 as per request, using #2566 as a guide.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by slester** [comment]
Signed.

**Comment by PoppySeedPlehzr** [comment]
Ok to test

**Comment by osqueryer** [comment]
:no_good: The commit 61eb301ebcada4114463e6cffae1be16f2004f09 (Job results: [429](https://jenkins.osquery.io/job/osqueryPullRequestAudit/429/)) failed the code audit.

**Comment by osqueryer** [comment]
:-1: The commit 61eb301ebcada4114463e6cffae1be16f2004f09 (Job results: [3447](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3447/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 61eb301ebcada4114463e6cffae1be16f2004f09 (Job results: [451](https://jenkins.osquery.io/job/osqueryPullRequestBuild/451/)) failed one or more tests (**Windows**).

**Comment by PoppySeedPlehzr** [comment]
With the windows `tools\make-win64-dev-env.bat` script you should have a plugin for Visual Studio that will clang-format everything. Inside of your table, hit Ctrl-A, then Ctrl-r+Ctrl-f to run the formatter and re-commit. Looks like there's a few pieces missing:
```
diff --git a/osquery/tables/networking/etc_services.cpp b/osquery/tables/networking/etc_services.cpp
index 26a5ace..1db87a9 100644
--- a/osquery/tables/networking/etc_services.cpp
+++ b/osquery/tables/networking/etc_services.cpp
@@ -29,11 +29,10 @@ namespace tables {
 
 #ifndef WIN32
 fs::path kServicesProtocols = "/etc/services";
-else
-fs::path kServicesProtocols = (getSystemRoot() / "system32\\drivers\\etc\\services");
+else fs::path kServicesProtocols =
+    (getSystemRoot() / "system32\\drivers\\etc\\services");
 #endif
 
-
 QueryData parseEtcServicesContent(const std::string& content) {
   QueryData results;
```

**Comment by PoppySeedPlehzr** [comment]
With the windows `tools\make-win64-dev-env.bat` script you should have a plugin for Visual Studio that will clang-format everything. Inside of your table, hit Ctrl-A, then Ctrl-r+Ctrl-f to run the formatter and re-commit. Looks like there's a few pieces missing:
```
diff --git a/osquery/tables/networking/etc_services.cpp b/osquery/tables/networking/etc_services.cpp
index 26a5ace..1db87a9 100644
--- a/osquery/tables/networking/etc_services.cpp
+++ b/osquery/tables/networking/etc_services.cpp
@@ -29,11 +29,10 @@ namespace tables {
 
 #ifndef WIN32
 fs::path kServicesProtocols = "/etc/services";
-else
-fs::path kServicesProtocols = (getSystemRoot() / "system32\\drivers\\etc\\services");
+else fs::path kServicesProtocols =
+    (getSystemRoot() / "system32\\drivers\\etc\\services");
 #endif
 
-
 QueryData parseEtcServicesContent(const std::string& content) {
   QueryData results;
```

It also looks like you didn't pull the spec file up. You'll want to me the `specs\posix\etc_services.table` file to `specs\etc_services.table` alongside your changes

**Comment by osqueryer** [comment]
:no_good: The commit 0aea70b4ee0634beaa292003a43b3bf7ff03fe0e (Job results: [430](https://jenkins.osquery.io/job/osqueryPullRequestAudit/430/)) failed the code audit.

**Comment by osqueryer** [comment]
:-1: The commit 0aea70b4ee0634beaa292003a43b3bf7ff03fe0e (Job results: [3448](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3448/)) failed one or more tests (**OS X/Linux**).

**Comment by PoppySeedPlehzr** [comment]
Huh..

Looks like it didn't quite get there:
```
diff --git a/osquery/tables/networking/etc_services.cpp b/osquery/tables/networking/etc_services.cpp
index 0478ee5..1db87a9 100644
--- a/osquery/tables/networking/etc_services.cpp
+++ b/osquery/tables/networking/etc_services.cpp
@@ -30,7 +30,7 @@ namespace tables {
 #ifndef WIN32
 fs::path kServicesProtocols = "/etc/services";
 else fs::path kServicesProtocols =
-  (getSystemRoot() / "system32\\drivers\\etc\\services");
+    (getSystemRoot() / "system32\\drivers\\etc\\services");
 #endif
```

Do you have the [clang-format plugin for visual studio](http://clang.llvm.org/docs/ClangFormat.html#visual-studio-integration) installed?

File: osquery/tables/networking/etc_services.cpp [hunk 1]
```diff
@@ -21,9 +22,17 @@
...
 #include "osquery/core/conversions.h"
 
+#define fs boost::filesystem
+
 namespace osquery {
 namespace tables {
 
+#ifndef WIN32
+fs::path kServicesProtocols = "/etc/services";
+else fs::path kServicesProtocols =
```
Comment (PoppySeedPlehzr) [inline osquery/tables/networking/etc_services.cpp]: Did you lose a `#` here? I think this should this be:
```
#ifndef WIN32
fs::path kServicesProtocols = ...
#else
fs::path kServicesProtocols = ...
#endif
```

**Comment by slester** [comment]
Sorry for all of the trouble. I'm just using `vim`, and I couldn't quite get the testing to work locally, so I tried to use your patch directly. It must have overzealously replaced the spacing. I'll try fixing that and repushing. Sorry again. :cry:

**Comment by osqueryer** [comment]
:-1: The commit 0aea70b4ee0634beaa292003a43b3bf7ff03fe0e (Job results: [452](https://jenkins.osquery.io/job/osqueryPullRequestBuild/452/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:no_good: The commit 88c51878d904648b8d3fcc9ee1b2952ce991ddda (Job results: [431](https://jenkins.osquery.io/job/osqueryPullRequestAudit/431/)) failed the code audit.

**Comment by osqueryer** [comment]
:-1: The commit 88c51878d904648b8d3fcc9ee1b2952ce991ddda (Job results: [453](https://jenkins.osquery.io/job/osqueryPullRequestBuild/453/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ee2ae56ea7aee483209bc8e691eccc13bcb95be0 (Job results: [3451](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3451/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 3750fa35fe34fd65a4bba74b8fdd88b8d0db5ece (Job results: [456](https://jenkins.osquery.io/job/osqueryPullRequestBuild/456/)) failed one or more tests (**Windows**).

**Comment by slester** [comment]
Is there a way to rerun the Windows test? Looking at the output, it looks like a Jenkins problem. Thanks.

**Comment by PoppySeedPlehzr** [comment]
Ok to test

**Comment by slester** [comment]
@PoppySeedPlehzr Thank you!

**Comment by theopolis** [comment]
FYI, https://github.com/facebook/osquery/pull/2595/files#diff-8905a8dbb701930cc3a62f10de9f05b9R25 should use the
```cpp
using fs = boost::filesystem;
```

**Comment by slester** [comment]
Thanks for the comment. How do I go about fixing this? It's also in the PR I based mine on (seeing as how this is my first PR for this project): https://github.com/facebook/osquery/pull/2590/files#diff-c9854c2bf46424e7eb4a4c1cfdab7d69R25

Should I submit a PR that fixes both/all of this?

**Comment by theopolis** [comment]
It's not urgent :) but sending a PR that fixes both would be epic!

**Pull request merged**